examples/protocols/memory/gddr7/gddr7-src.stanza

```diff
@@ -0,0 +1,649 @@
+
+  val box-params = BoxSymbolParams(
+    show-grid = false
+  )
+
+  val box = BoxSymbol(self, params = box-params)
+
+  for (c in [channel-A channel-B channel-C channel-D], i in 0 to false) do :
+    for j in 0 to 10 do :
+      set-bank(c, DQp[j][i])
+      set-side(Left, DQp[j][i])
+    set-bank(c, RCK_c[i])
+    set-bank(c, RCK_t[i])
+    set-bank(c, WCK_c[i])
+    set-bank(c, WCK_t[i])
+    set-side(Right, RCK_c[i] RCK_t[i] WCK_c[i] WCK_t[i])
+    for j in 0 to 5 do :
+      set-bank(c, CA[j][i])
+      set-side(Right, CA[j][i])
+    set-bank(c, DQE[i])
+    set-side(Right, DQE[i])
+    set-bank(c, ERR[i])
+    set-side(Right, ERR[i])
+
+  set-bank(ctl-r, self.ZQ_AB self.ZQ_CD, self.RESET_n)
+  set-side(Left, self.ZQ_AB self.ZQ_CD, self.RESET_n)
+  set-bank(pwr, self.VSS self.VDD self.VDDQ self.VPP)
+  set-side(Down, self.VSS)
+  set-side(Up, self.VDD self.VDDQ self.VPP)
+
+
+  assign-symbols(
+    ctl-r     => box,
+    pwr       => box,
+    channel-A => box,
+    channel-B => box,
+    channel-C => box,
+    channel-D => box,
+  )
+
+  val lp = create-landpattern(FBGA266())
+  assign-landpattern(lp)
+
+  for i in 0 to 4 do :
+    diff-pin-model(self.RCK_c[i], self.RCK_t[i], delay = typ(1.0e-15) loss = typ(0.1))
+    diff-pin-model(self.WCK_c[i], self.WCK_t[i], delay = typ(1.0e-15) loss = typ(0.1))
+
+  for i in 0 to 4 do :
+    for j in 0 to 10 do :
+      pin-model(self.DQp[j][i]) = PinModel(typ(1.0e-15), typ(0.1))
+    for j in 0 to 5 do :
+        pin-model(self.CA[j][i]) = PinModel(typ(1.0e-15), typ(0.1))
+    pin-model(self.DQE[i]) = PinModel(typ(1.0e-15), typ(0.1))
+    pin-model(self.ERR[i]) = PinModel(typ(1.0e-15), typ(0.1))
+  pin-model(self.RESET_n) = PinModel(typ(1.0e-15), typ(0.1))
+
+public pcb-module mem-device :
+  port io : gddr7()
+  port pwr : power
+
+  public inst mem : jsl/examples/protocols/memory/gddr7/gddr7-src/mem-component
+
+  for i in 0 to 4 do :
+    for j in 0 to 10 do :
+      topo-net(io.data[i].DQ[j] mem.DQp[j][i])
+    topo-net(io.data[i].RCK.P mem.RCK_t[i])
+    topo-net(io.data[i].RCK.N mem.RCK_c[i])
+    topo-net(io.data[i].WCK.P mem.WCK_t[i])
+    topo-net(io.data[i].WCK.N mem.WCK_c[i])
+    topo-net(io.data[i].DQE   mem.DQE[i])
+    topo-net(io.data[i].ERR   mem.ERR[i])
+    for j in 0 to 5 do :
+      topo-net(io.data[i].CA[j] mem.CA[j][i])
+
+  topo-net(io.control.RESET_n mem.RESET_n)
+  topo-net(io.control.ZQ_AB mem.ZQ_AB)
+  topo-net(io.control.ZQ_CD mem.ZQ_CD)
+
+  for p in pins(mem.VDDQ) do :
+    net (pwr.V+ p)
+
+  for p in pins(mem.VDD) do :
+    net (pwr.V+ p)
+
+  for p in pins(mem.VSS) do :
+    net (pwr.V- p)
+
+
+val ic-pkg-body = PackageBody(
+  width = 24.0 +/- 0.1,
+  length = 24.0 +/- 0.1,
+  height = 0.71 +/- [0.06, 0.0]
+)
+
+val ic-bga-pkg = BGA(
+  num-leads = 900,
+  rows = 30,
+  columns = 30,
+  lead-diam = 0.25,
+  pitch = 0.75,
+  package-body = ic-pkg-body,
+  density-level = DensityLevelB
+)
+
+defn lookup-bga-row-name (idx:Int, table:Tuple<Char>) :
```

Comment:
There is a lot of code related to letter row lookup etc in `numbering.stanza`:

https://github.com/JITx-Inc/jsl/blob/88bf97f505ed0ebd74da9c6d418fd3d6eb102b9f/src/landpatterns/numbering.stanza#L471

Are we sure we can't reuse that ?